dlt/common/libs/pyarrow.py

```diff
@@ -1047,3 +1047,25 @@ class NameNormalizationCollision(ValueError):
     def __init__(self, reason: str) -> None:
         msg = f"Arrow column name collision after input data normalization. {reason}"
         super().__init__(msg)
+
+
+def add_arrow_metadata(
```

Comment:
there are several function in that module that will remove this metadata but don't have to. basically all that recreate schemas from list of columns and arrays. also method that creates arrow schema from columns does not set this metadata. maybe it should? at least we need to mark it as tech debt

Reply:
Ok, I will not do this now, but have started a ticket for initial transformation release follow up work: https://github.com/dlt-hub/dlt/issues/2827